service-front/module/Application/src/Controller/DonorFlowController.php

```diff
@@ -95,15 +97,57 @@ public function howWillDonorConfirmAction(): ViewModel|Response
                             $uuid,
                             $data
                         );
-                        return $this->redirect()->toRoute("root/donor_details_match_check", ['uuid' => $uuid]);
+                        $returnRoute = "root/donor_details_match_check";
                     }
+                    return $this->redirect()->toRoute($returnRoute, ['uuid' => $uuid]);
                 }
             }
         }
 
         return $view->setTemplate($templates['default']);
     }
 
+    public function whatIsVouchingAction(): ViewModel|Response
+    {
+        $view = new ViewModel();
+        $uuid = $this->params()->fromRoute("uuid");
+        $detailsData = $this->opgApiService->getDetailsData($uuid);
+        $view->setVariable('details_data', $detailsData);
+
+        if ($this->getRequest()->isPost()) {
+            $formData = $this->getRequest()->getPost()->toArray();
+            if ($formData['confirm_vouching'] == 'yes') {
+                /**
+                 * @psalm-suppress ArgumentTypeCoercion
+                 */
+                $pdf = $this->siriusApiService->sendPdf($detailsData, "DLP-VOUCH-INVITE", $this->request);
```

Comment:
If you use `getRequest` you can avoid needing the Psalm suppression, since it will always return `Laminas\Http\Request`
```suggestion
                $pdf = $this->siriusApiService->sendPdf($detailsData, "DLP-VOUCH-INVITE", $this->getRequest());
```

Reply:
Also I'd suggest moving the template IDs to either an enum or class constants, rather than having them in code here, so that you can do something like:
```php
$this->siriusApiService->sendPdf($detailsData, SiriusDocument::VouchInvitation, $this->request);
```
That isolates the Sirius configuration into a centralised file in case it needs to be changed, and makes the intention of the document clearer (not so much a problem here, but I'm guessing "DLP-ID-PO-D" doesn't mean much to most people working Paper ID 😁)